src/result.test.ts

```diff
@@ -1980,6 +2250,20 @@ describe("AsyncResult", () => {
 				Result.assertError(result);
 				expect(result.error).toBeInstanceOf(CustomError);
 			});
+
+			it("transforms thrown errors when an errorTransform function is defined", async () => {
```

Comment:
I think the same comments of the sync-variant of `mapCatching` apply here as well